### ISSUE TITLE
AI-PFR Potential Opportunities

### ISSUE THREAD

**Issue by sarahcleary** [APSIMInitiative/ReferencePanel#13]
AI to consider the following five areas of potential future focus/priority

    Developing horticultural/perennial tree crop functionality in APSIM.
    Climate change response functionality in APSIM.
    Developing multi-functional landscape capability in APSIM.
    Precision agriculture.
    Considering high performance computing needs and potential implications. 

Recommendation - First step would be to collate what everyone is doing. 

To ensure APSIM Infrastructure can deal with the high performance computer. Keith to start Github repository

**ACTION** RP agreed to review each of these areas as part of the review of the Development Plan. Each home org would provide interest, synergy as part of the process of developing the plan for 19/20.

Link to Action: https://github.com/APSIMInitiative/ReferencePanel/issues/7

**Comment by hol353** [comment]
The CSIRO APSIM project has allocations and operating to support some of these activities. CSIRO staff and PFR staff undertake two face-to-face sprints a year on these and various APSIM Next Generation maintenance issues.

**Developing horticultural/perennial tree crop functionality in APSIM.**
Build a perennial model (e.g. grape vine) for APSIM Next Generation - Neil Huth 30 days
This is in development plan under 'Develop a Grape Model for APSIM Next Gen'. Perhaps this should merge into activity 'Develop Simple Perennial Fruit tree model in PMF (APSIM next gen)'. @HamishBrownPFR ?

**Climate change response functionality in APSIM.**
Unsure that this means precisely but CSIRO does a lot of climate variability and change work that uses APSIM. I'm unsure where the linkages are though but it would be good to have some shared project work.

**Considering high performance computing needs and potential implications.**
Add a machine learning capability to APSIM - Neil Huth 50 days. It is under APSIM Infrastructure in the development plan. This is sort of related to high performance computing and is something that Neil and Hamish will work on together. In the past I have also worked with Edmar Teixeira and Eric Burgueño from PFR on PFR's HPC compute platform. This work will likely be ongoing from time to time.

**Comment by sarahcleary** [comment]
comments received from Val/AgR:


•	Developing horticultural/perennial tree crop functionality in APSIM.
o	I don’t think that there is anything that we can directly contribute here or directly collaborate on but highly supportive of the work. There are follow-on effects:
	In order to do tree row crops there will need to be some updating in Micromet. We have a historical interest in Micromet (Neil Huth (CSIRO) and I developed it) and I would prefer that Neil and I did the updating needed for all sorts of reasons. We do not have funding for this activity however.
	We have a Brazilian PhD student coming for a 6-month visit in October. He will be working on agroforestry (eucalypts and pastures) and there are loose links there.
	The ability of APSIM to deal with more horticultural crops is obviously of interest to our desire to be able to simulate diverse landscapes. Ultimately we will want to graze sheep or alpacas or something under those trees and produce honey from the red clover being used as a substitute for N fertiliser and start dealing with trade-offs and ecosystem services (links to third dot point).
•	Climate change response functionality in APSIM.
o	I think that PFR here mean elevated CO2 effects on stomatal function and plant processes. I don’t know of anything going on in that area but we are interested (nothing active that I know about but I need to check in with Mark Lieffering) in the flow-on effects to soil C and N processes (progressive nutrient limitation).
•	Developing multi-functional landscape capability in APSIM.
o	Yes of interest (see above). Nothing funded as yet (there is a just-concluded Catalyst project on viticulture and sheep but that did not include modelling) but a key direction that we need to be able to head in.
•	Precision agriculture.
o	Kind of a broad field so difficult to provide a short answer. We have done work with Lincoln Agritech on precision fertiliser but that finished with the project (and when Iris left). 
•	Considering high performance computing needs and potential implications.
o	Something of interest to many but also something that is really hard to collaborate on other than sharing tools and tips. The issue here is that different institutions are locked into different platforms (AgResearch & Azure, CSIRO & their own compute platform, ditto with PFR, UQ and USQ) with differing requirements and very high barriers to using resources outside those set up by your own institution (some firewall type issues but also cost). There are design features of APSIM that can make using HPC platforms in general easier or harder but I suspect we are on top of those already. A further issue with HPCs is the soil and weather data. NZ does not have the soil tool box in APSIM that Australia, Asia/Africa, USA (well – Iowa!), … have and the weather data is not freely available (or not freely shareable either) and that limits our ability to do national level studies.

**Comment by sarahcleary** [comment]
comments received from UQ:
•	Developing horticultural/perennial tree crop functionality in APSIM.
UQ has a horticultural modelling group (with Jim Hanan as lead) working on 3D architectural models (L-System style) eg. macadamia.  At present this is not linked with APSIM, but it could be if there was interest to do that

•	Climate change response functionality in APSIM.
UQ has developed and published (recent Nature Plants paper) a biochemical level photosynthesis module for C3 and C4 plants that has been linked with APSIM and tested successfully on sorghum and wheat validation data sets.  This enables comprehensive simulation of enhanced CO2 and temperature/vpd effects.  We have utilised climate change projection weather data sets (held with Bangyou Zheng – CSIRO) in simulation studies.  We have also incorporated effects of heat shock stress on sorghum and wheat (seed set, senescence) using simple index approaches and published this with David Lobell (Global Change Biol paper).

•	Considering high performance computing needs and potential implications.
We have a supercomputing centre at UQ but have not progressed this linkage at this point in time.  Given intent to press on linking genomics advances with breeding via Crop Growth modelling in APSIM and optimisation approaches, we will be exploring computing requirements for that research over the next year.  There may be implications around this in relation to software development eg optimisation shells

**Comment by sarahcleary** [comment]
@APSIMInitiative/reference-panel - awaiting USQ and DAF's input.

**Comment by sarahcleary** [comment]
Email reminder sent to DAF and USQ for input.

**Comment by sarahcleary** [comment]
Comments from DAF:


All areas indicated in the AI-PFR meeting are relevant to near term and long-term future requirements. 

•	It use for perennial crops will be valuable for developing agro forestry systems where shortage of water will limit cultivation of grain crops as well as horticultural crops requiring a lot of water. Even without that scenario having arrived, it will be helpful in diversifying our farming systems

•	APSIM will be increasingly used for understanding and developing strategies for climate change and as a decision support tool. APSIM will need to be made more versatile to address these needs. In relation to climate change response functionality, we’re planning to do some work with UQ regarding the photosynthesis work that Alex Wu has done.  This work is waiting on the conversion of the sorghum model from the old framework into the new one

•	Multifunctional landscapes will be critical for meeting varied food needs while maintaining sustainability. Currently APSIM does not address this adequately

•	High performance computing (HPC)  is a must have requirement if we are to use APSIM for crop improvement and optimising genotype x environment x management interactions. Especially when more gene to phenotype models become in vogue. We don’t have the capacity to provide this ourselves as it is expensive. We are currently reliant on other groups to utilise these systems

**Comment by sarahcleary** [comment]
From @keith-pembleton (USQ):  I have a coursework masters student interested in doing perennial horticultural modelling for his research project (I am already talking with Hamish about this). 
We maybe have a slight interest in the climate change one, but that is more for historical reasons rather than any current research activities.
